pkg/ibmcsidriver/node.go

```diff
@@ -171,6 +171,11 @@ func (csiNS *CSINodeServer) NodeUnpublishVolume(ctx context.Context, req *csi.No
 		return nil, commonError.GetCSIError(ctxLogger, commonError.NoTargetPath, requestID, nil)
 	}
 
+	//Lets try to put lock at targetPath level. If we are processing same target path lets wait for other to finish.
+	//This will not hold other volumes and target path processing.
+	csiNS.mutex.Lock(targetPath)
```

Comment:
why we need lock for umount

Reply:
We need to avoid mount and umount race condition for same targetPath. We have also referred other providers and found GCP that this is recommended

https://github.ibm.com/alchemy-containers/armada-storage/issues/5245#issuecomment-69052970